### PR TITLE
feat: Add ICS Feed support

### LINKED ISSUE #3167: [CAL-2979] ICS feed support for checking against double bookings
### Is your proposal related to a problem?

I teach a language on italki. When people book lessons, the lessons event get dynamically updated in an .ics file calendar that's hosted through http. I use software (Thunderbird on my computer and ICSx5 on my Android smartphone) to see this calendar along my usual calendar (in CalDAV).

[Cal.com](http://Cal.com) has the ability to check my CalDAV for events that are present when someone want to book a meeting. Sadly, if someone booked a lesson on italki, the lesson will only show in the read-only ICS calendar. Therefore, someone could book a meeting on my [Cal.com](http://Cal.com) page even though I'm already booked on italki.

I have no abilities to automatically import the ICS feed in my CalDAV calendar for [Cal.com](http://Cal.com) to see when I'm busy.

### Describe the solution you'd like

I would click on addons, click on ICS calendar, enter the link and press okay. [Cal.com](http://Cal.com) would check this one and my CalDAV calendar and prevent double bookings.

It would be awesome if [Cal.com](http://Cal.com) had an app to read the ICS calendar (the same way it reads my CalDAV calendar) when someone is trying to book a meeting. ICS calendars are very easy to set up (just a link) and I believe there wouldn't be much to be changed from the CalDAV [Cal.com](http://Cal.com) support app code.

Integrated support for ICS feeds is common in most calendar solution, so it would only make sense [Cal.com](http://Cal.com) also does. (IOS, Google and Thunderbird all support it. Sadly, not automatic import, because it doesn't really make sense except for my use case).

### Describe alternatives you've considered

I've added the calendar as a subscription to my CalDAV calendar (hosted at Infomaniak), but [cal.com](http://cal.com) is totally incoherent when it comes to seeing them or not (sometimes it does and I can't book times where I'm unavailable and sometimes it doesn't). As such, I think it would work better if this was part of [Cal.com](http://Cal.com) features.

addedcalendars

### Additional context

Thanks for your answer and work on [Cal.com](http://Cal.com)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2979](https://linear.app/calcom/issue/CAL-2979/ics-feed-support-for-checking-against-double-bookings)</sub>

### PR DESCRIPTION
**Pull request by mogery**
## What does this PR do?

Adds an "ICS Feed" app to the app store that allows the linking of ICS Feeds for usage via `getAvailability`

Fixes #3167 
/claim #3167 

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

Criteria:
  - "ICS Feed" can be installed from the App Store
  - Setup screen pops up, ICS feed URLs can be added and saved
  - ICS feed calendars appear in the system and are toggleable to be used for conflict checking
  - ICS feeds are actually used for conflict checking

There's an ICS feed of German holidays in #3167 that I used for testing.

### CODE REVIEW

**Comment by mogery** [comment]
Couldn't revert the yarn.lock change since this updates the workspace structure.

**Comment by PeerRich** [comment]
awesome initiative 

this will come handy for many people using ics

File: packages/app-store/ics-feedcalendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    return 0;
+  }
+};
+
+const applyTravelDuration = (event: ICAL.Event, seconds: number) => {
+  if (seconds <= 0) return event;
+  // move event start date back by the specified travel time
+  event.startDate.second -= seconds;
+  return event;
+};
```
Comment (mogery) [inline packages/app-store/ics-feedcalendar/lib/CalendarService.ts:37]: not sure how much this is actually needed? it certainly doesn't hurt though

**Review by emrysal — APPROVED** [review]
Tested, code wise this looks really good too; Happy to approve - Thank you!

**Comment by saschafoerster** [comment]
So cool! 😎

**Comment by shafy** [comment]
Really cool! Is this already live on the cal.com version yet? I can't seem to find it...

**Comment by penyuan** [comment]
@emrysal @mogery Same here, how do I find this feature on cal.com????

**Comment by penyuan** [comment]
I'm getting the same error as @wivaku as described in issue #3167. I shared an Outlook 365 calendar under my control with full view permissions, with URL in the form of `https://outlook.office365.com/owa/calendar/<...>/<...>/calendar.ics`, and copied it into the "ICS Feed" app. And I got an error saying: 

`No handlers found for ics-feed`

@mogery @emrysal and other developers: Can you please advise? Thank you!!

**Pull request merged**